### ISSUE TITLE
issue with .net core 5

### ISSUE THREAD

**Issue by idhasitha** [dotnet-architecture/eShopOnWeb#484]
I use this architect and solution for my project and this work well 
but today when I upgrade the solution to .net core 5 I have an issue with context 
I used the BaseEntity and inherit it to other entity and use generic crud operation with T entity 

public class BaseEntity
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int Id { get; set; }
    }

public class Survey : BaseEntity
    {
        public int ShareType { get; set; }
        public int SharedId { get; set; }

}

public List<T> ListAll<T>() where T : BaseEntity
        {
            return DbContext.Set<T>().ToList();
        }

Error 
: 

Message: "Cannot create a DbSet for 'BaseEntity' because this type is not included in the model for the context."
	StatusCode: 500

**Comment by fingers10** [comment]
Try to use fluent configuration to configure id for each individual entity instead of using annotation attribute in Base Entity

**Comment by idhasitha** [comment]
@fingers10 

then we are not able to use a generic operation like this 

 public List<T> ListAll<T>() where T : BaseEntity
        {
            return surveyDbContext.Set<T>().ToList();
        }

**Comment by fingers10** [comment]
try ` surveyDbContext.Set<T>().ToList()` instead of `surveyDbContext.Set().ToList()`

**Comment by idhasitha** [comment]
@fingers10  but we do need to define a T class 

public interface IAsyncRepository<T> where T : BaseEntity, IAggregateRoot
    {
        Task<T> GetByIdAsync(int id);
        Task<IReadOnlyList<T>> ListAllAsync();
        Task<IReadOnlyList<T>> ListAsync(ISpecification<T> spec);
        Task<T> AddAsync(T entity);
        Task UpdateAsync(T entity);
        Task DeleteAsync(T entity);
        Task<int> CountAsync(ISpecification<T> spec);
        Task<T> FirstAsync(ISpecification<T> spec);
        Task<T> FirstOrDefaultAsync(ISpecification<T> spec);
    }

**Comment by fingers10** [comment]
Yes you need to define. Please refer the repository in Infrastructure project.
